presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionProcessFactory.java

```diff
@@ -17,87 +17,135 @@
         this.planFragmentCodec = requireNonNull(planFragmentCodec, "planFragmentCodec is null");
         this.taskUpdateRequestCodec = requireNonNull(taskUpdateRequestCodec, "taskUpdateRequestCodec is null");
+        this.serverInfoCodec = requireNonNull(serverInfoCodec, "serverInfoCodec is null");
         this.taskManagerConfig = requireNonNull(taskManagerConfig, "taskManagerConfig is null");
+        this.systemConfig = requireNonNull(systemConfig, "systemConfig is null");
+        this.nodeConfig = requireNonNull(nodeConfig, "nodeConfig is null");
+        this.connectorConfig = requireNonNull(connectorConfig, "connectorConfig is null");
     }
 
-    public NativeExecutionTask createNativeExecutionTask(
+    public NativeExecutionProcess createNativeExecutionProcess(
             Session session,
             URI location,
             TaskId taskId,
             PlanFragment fragment,
             List<TaskSource> sources,
             TableWriteInfo tableWriteInfo)
+            throws IOException
     {
-        return new NativeExecutionTask(
-                session,
-                location,
-                taskId,
-                fragment,
-                sources,
-                httpClient,
-                tableWriteInfo,
-                executor,
-                updateScheduledExecutor,
-                taskInfoCodec,
-                planFragmentCodec,
-                taskUpdateRequestCodec,
-                taskManagerConfig);
+        return createNativeExecutionProcess(session, location, taskId, fragment, sources, tableWriteInfo, MAX_ERROR_DURATION);
+    }
+
+    public NativeExecutionProcess createNativeExecutionProcess(
+            Session session,
+            URI location,
+            TaskId taskId,
+            PlanFragment fragment,
+            List<TaskSource> sources,
+            TableWriteInfo tableWriteInfo,
+            Duration maxErrorDuration)
+    {
+        try {
+            return new NativeExecutionProcess(
+                    session,
+                    location,
+                    taskId,
+                    fragment,
+                    sources,
+                    tableWriteInfo,
+                    httpClient,
+                    executor,
+                    errorRetryScheduledExecutor,
+                    updateScheduledExecutor,
+                    taskInfoCodec,
+                    planFragmentCodec,
+                    taskUpdateRequestCodec,
+                    serverInfoCodec,
+                    maxErrorDuration,
+                    taskManagerConfig,
+                    systemConfig,
+                    nodeConfig,
+                    connectorConfig);
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
     }
 
     @PreDestroy
     public void stop()
     {
         coreExecutor.shutdownNow();
         updateScheduledExecutor.shutdownNow();
+        errorRetryScheduledExecutor.shutdown();
```

Comment:
others are calling shutdownNow, this calls shutdown

Reply:
good catch.